### ISSUE TITLE
"In future versions of macOS the tcl runtime will not be available by  default"

### ISSUE THREAD

**Issue by kainjow** [HexFiend/HexFiend#427]
If you run `tclsh` you'll get:
> WARNING: This version of tcl is included in macOS for compatibility with legacy software. 
In future versions of macOS the tcl runtime will not be available by 
default, and may require you to install an additional package.

I haven't yet heard TCL being removed from Sonoma, but it could happen after. If/when this does happen, we'll need to embed TCL.

**Comment by xyzzy-plugh-plover** [comment]
Will Tcl be updated to 8.6.13 in 2.17?

**Comment by kainjow** [comment]
No. Probably limit it to a smaller release, since I’m not sure if there are backwards-incompatible changes with 8.6. It’d need more testing,

**Comment by kainjow** [comment]
What would be neat is to be able to choose between the system Tcl and the built-in one. The most direct way would be to dynamically load the system Tcl.framework. This seems tricky though. Alternatively, loadable bundles could be added that themselves load the framework, and then at launch one is chosen based on a preference. This would allow for users to choose which one they'd want to use and experiment using the newer version, vs forcing the new version on everyone.

**Comment by kainjow** [comment]
Tcl 9.0 released! https://www.tcl-lang.org/software/tcltk/9.0.html?post=new
